server/fleet/app.go

```diff
@@ -657,12 +657,16 @@ type EnrollSecretSpec struct {
 
 const (
 	// tierBasic is for backward compatibility with previous tier names
-	tierBasic = "basic"
+	TierBasic = "basic"
```

Comment:
I think this should stay unexported, the goal was probably to make sure it doesn't get used anyhwere outside this package.

Reply:
Ah nevermind, just noticed that it was being used in `ee/server/licensing/licensing.go` and that the old code would replace the "trial" license with "premium" which we don't want... Maybe rename to `TierBasicDeprecated` just to make it clear?

Reply:
Yeah that's reasonable. I'll create a method like ForceUpgrade that'll internally check basic tier and upgrade if necessary.